src/lustre/lustreNodeGen.ml

```diff
@@ -1587,6 +1587,28 @@ and compile_node_decl gids is_function cstate ctx i ext inputs outputs locals it
       result :: glocals
     in List.fold_left over_generated_locals glocals gids.GI.subrange_constraints
   (* ****************************************************************** *)
+  (* (State Variables for) Generated Array Constraints                  *)
+  (* ****************************************************************** *)
+  in let glocals =
+    let over_generated_locals glocals (_, id, _, _) =
+      let ident = mk_ident id in
+      let index_types = compile_ast_type cstate ctx map (A.Bool dummy_pos) in
+      let over_indices = fun index index_type accum ->
+        let possible_state_var = mk_state_var
+          map
+          (node_scope @ I.reserved_scope)
+          ident
+          index
+          index_type
+          (Some N.KGhost)
+        in
+        match possible_state_var with
+        | Some state_var -> X.add index state_var accum
+        | None -> accum
+      in let result = X.fold over_indices index_types X.empty in
+      result :: glocals
+    in List.fold_left over_generated_locals glocals gids.GI.array_constraints
+  (* ****************************************************************** *)
```

Comment:
Why do you use ad hoc code for these variables? You should add them to the field `locals` in `generatedIdentifiers` at the same time that you add the generated equation.

Reply:
Addressed